### DISCUSSION TITLE
RCON not able to connect after restart of minecraft-server

### DISCUSSION THREAD

**Discussion by tbe2eci** [itzg/docker-minecraft-server#2997] [Q&A] [answered]
Hi,
I am quite new to all this, so I have likely missed something obvious. I hope you can help to pinpoint what I have been missing.
I have setup 2 docker-containers, itzg/minecraft-server and itzg/rcon.
They are both deployed with container-station on my QNAP NAS, and they work as expected.
When I deployed rcon I used environment-variables to adapt to the generated  rcon.password in server.properties on the minecraft-server.
However, when I restart the minecraft-server, it generates a new rcon.password and rcon is no longer able to login to the server.
I need to spawn a new rcon-container with a new rcon.password in the environment-variables.

I totally see the point with generating an rcon-password when the Minecraft-server is created, but I'm not sure why it's done at every restart. Can I prevent that in any way?
My goal is to not being forced to spawn a new rcon-container everytime I restart the Minecraft-server.
Any advice?

I have found a number of discussions in the same area, but couldn't find any that was answering this question.

**Comment by itzg** [comment]
Somewhere in the docs I state that you must set a specific RCON password and not use the default generated one.

**Reply by tbe2eci** [reply]
Thank you for your quick reply!
I tried that as well, but the rcon.password was still overwritten.
Right now I realized that perhaps I need to set that before I spawn the server. In my attempt I modified it while the server was running and then I restarted it.
Is that how it works (that I need to modify it before I spawn the server) ?

**Reply by itzg** [reply, marked as answer]
Refer to https://docker-minecraft-server.readthedocs.io/en/latest/configuration/server-properties/#rcon

**Comment by tbe2eci** [comment]
Yes, it works now. Thanks a lot for leading a blind man :-)
When starting the Minecraft-server I now included the environment-variable RCON_PASSWORD and voila, it worked.
(It did not work with modifying the server.properties directly as I were trying out first)

Next step is now to create a new server with this setting, and copy the content of the old server to it.
I have seen a number of guides describing how to do that, so I think that will be quite straight forward.
